src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder(String.format("Report for period %s - %s",
+                dateFrom, dateTo));
+
+        String[] foundData;
+        int[] salaries = new int[names.length];
+        for (String d : data) {
+            String date = d.substring(0, 10);
+            if (compareDates(dateFrom, date, dateTo)) {
+                for (int i = 0; i < names.length; i++) {
```

Comment:
```suggestion
        for (String name : names) {
            int salary = 0;
            for (String line : data) {
```
append your StringBuilder inside outer loop (after inner)